lib/ansible/modules/network/vyos/vyos_interface.py

```diff
@@ -204,6 +204,8 @@ def map_obj_to_commands(updates):
                     value = w.get(item)
 
                     if value and value != obj_in_have.get(item):
+                        if item == 'description':
+                            value = "\'" + str(value) + "\'"
```

Comment:
Just a note, str is not safe to use like this if value can contain non-ascii characters.  let me know if that's the case and I can help you test that case and fix it.